### ISSUE TITLE
macro bug, seems to work once and then get stuck

### ISSUE THREAD

**Issue by ldemailly** [grol-io/grol#223]
```go

unless(5 > 11, println("macro test 1: lower ok"), println("BUG: not lower"))

unless(10 > 5, println("BUG: not greater"), println("macro test 2: greater")) // should output "macro test: greater" and not again "lower ok"
```

actually currently prints
```
macro test: lower ok
macro test: lower ok
```

also unable to get 
```go
NoErr = macro(expr,a,b) {
    quote(
		r = catch(unquote(expr))
	    if r.err {
			error("unexpected error:", r.value)
			unquote(a)
		} else {
			unquote(b)
		}
	)
}
```
to work... complains about expr being unknown...

**Comment by ldemailly** [comment]
it's basically #20

**Comment by ldemailly** [comment]
fixed in #225
